sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -178,11 +195,16 @@ object NestedColumnAliasing {
             nestedFieldToAlias
               .map { case (nestedField, _) => totalFieldNum(nestedField.dataType) }
               .sum < totalFieldNum(attr.dataType)) {
-          Some(attr.exprId -> nestedFieldToAlias)
+          Some((attr.exprId, nestedFieldToAlias))
         } else {
           None
         }
       }
+      .groupBy(_._1) // To fix same ExprId mapped to different attribute instance
```

Comment:
Is this an existing issue? Could you show us a query to reproduce this?

Reply:
https://github.com/apache/spark/pull/28898/files#diff-957112380b0a2ef014abc8227d0b70acR479-R496
In this case, the attribute is `name`, but they have different qualifiers. So the `groupBy` isn't quite working here. Then at the latter `flatMap`, it can cause collision on the same `ExprId`.

Reply:
You meant this fix is only for the Window case?

Reply:
Not necessarily only for the `Window` case, but by adding `Window/Filter/Sort`, this error can be surfaced more easily.

Reply:
I don't think I can reproduce it without adding `Window/Filter/Sort`, but generally if someone writes a complicated query, this might be triggered even without adding `Window/Filter/Sort`.

Reply:
Simplified the logic above to do grouping by `ExprId` and updated the deduplication logic.